pkg/packages/manager.go

```diff
@@ -65,11 +71,67 @@ func (mc *ManagerContext) getRegistry(values map[string]interface{}) string {
 func processInitializing(mc *ManagerContext) bool {
 	mc.Log.Info("New installation", "name", mc.Package.Name)
 	mc.Package.Status.Source = mc.Source
-	mc.Package.Status.State = api.StateInstalling
+	mc.Package.Status.State = api.StateInstallingDependencies
 	mc.RequeueAfter = retryNow
 	return true
 }
 
+func processUpdating(mc *ManagerContext) bool {
+	mc.Log.Info("Updating package ", mc.Package.Name)
+	mc.Package.Status.State = api.StateInstallingDependencies
+	mc.RequeueAfter = retryNow
+	return true
+}
+
+func processInstallingDependencies(mc *ManagerContext) bool {
+	mc.Log.Info("Installing dependencies", "chart", mc.Source)
+	dependencies, err := mc.Bundle.GetDependencies(mc.Version)
+	if err != nil {
+		mc.Package.Status.Detail = fmt.Sprintf("invalid package bundle. (%s@%s bundle: %s)", mc.Package.Name, mc.Version, mc.Bundle.Name)
+		mc.Log.Info(mc.Package.Status.Detail)
+		mc.RequeueAfter = retryLong
+		return true
+	}
+	pkgs, err := mc.PackageClient.GetPackageList(mc.Ctx, mc.Package.Namespace)
```

Comment:
some more descriptive name would help here like installedPackages

Reply:
for `GetPackageList` or `pkg`? We don't filter or know about the state at this point.

Reply:
for pkg